controller_manager:
  ros__parameters:
    update_rate: 1000 # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

trajectory_controller:
  ros__parameters:
    joints:
      - FL_hip_joint
      - FL_thigh_joint
      - FL_calf_joint
      - FR_hip_joint
      - FR_thigh_joint
      - FR_calf_joint
      - RL_hip_joint
      - RL_thigh_joint
      - RL_calf_joint
      - RR_hip_joint
      - RR_thigh_joint
      - RR_calf_joint
    
    command_interfaces:
      - effort
    
    state_interfaces:
      - position
      - velocity

    allow_partial_joints_goal: True
    action_monitor_rate: 20.0
    state_publish_rate: 50.0

    gains:
      FL_hip_joint: {p: 100.0, i: 0.0, d: 5.0, ff_velocity_scale: 1.0}
      FL_thigh_joint: {p: 300.0, i: 0.0, d: 8.0, ff_velocity_scale: 1.0}
      FL_calf_joint: {p: 300.0, i: 0.0, d: 8.0, ff_velocity_scale: 1.0}
      FR_hip_joint: {p: 100.0, i: 0.0, d: 5.0, ff_velocity_scale: 1.0}
      FR_thigh_joint: {p: 300.0, i: 0.0, d: 8.0, ff_velocity_scale: 1.0}
      FR_calf_joint: {p: 300.0, i: 0.0, d: 8.0, ff_velocity_scale: 1.0}
      RL_hip_joint: {p: 100.0, i: 0.0, d: 5.0, ff_velocity_scale: 1.0}
      RL_thigh_joint: {p: 300.0, i: 0.0, d: 8.0, ff_velocity_scale: 1.0}
      RL_calf_joint: {p: 300.0, i: 0.0, d: 8.0, ff_velocity_scale: 1.0}
      RR_hip_joint: {p: 100.0, i: 0.0, d: 5.0, ff_velocity_scale: 1.0}
      RR_thigh_joint: {p: 300.0, i: 0.0, d: 8.0, ff_velocity_scale: 1.0}
      RR_calf_joint: {p: 300.0, i: 0.0, d: 8.0, ff_velocity_scale: 1.0}
